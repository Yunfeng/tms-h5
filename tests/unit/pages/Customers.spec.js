import Component from '@/pages/Customers.vue'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

import $ from 'jquery'


const localVue = createLocalVue()
localVue.use(Vuex)


describe('Customers.vue', () => {

  let wrapper
  const routePath = '/customers'

  let actions
  let store

  beforeEach(() => {

    actions = {
      showLoading: jest.fn()
    }

    store = new Vuex.Store({
      actions
    })
    
    const $route = {
  		path: routePath
    }

    wrapper = shallowMount(Component, {
      store,
      localVue,
    	mocks: {
    		$route
      },
      stubs: ['router-link', 'router-view']
    })
  });

  // test方法是it的别名之一
  test('matches snapshot', () => {
    // wrapper = shallowMount(Component, {
    //   stubs: ['router-link', 'router-view']
    // })
  //   // expect(vm.$el).toMatchSnapshot()
  //   // expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('mount users', () => {
    expect(wrapper.vm.$route.path).toBe(routePath)
    expect(wrapper.isVueInstance()).toBeTruthy()
  });


  it('has the expected html structure', () => {
    // const wrapper = shallowMount(Component)

    expect(wrapper.find('#customers').exists()).toBeTruthy()
    expect(wrapper.find('table tbody tr').length).toBeUndefined()
  })

})
