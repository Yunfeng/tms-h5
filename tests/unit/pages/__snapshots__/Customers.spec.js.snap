// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customers.vue matches snapshot 1`] = `
<div id="customers">
  <div class="card">
    <div class="card-header bg-info text-white">
      客户信息
    </div>
    <div class="card-body">
      <form id="frmSearchUser" class="form-inline"><input type="hidden" name="sc.pageNo" value="1">
        <div class="form-group"><input type="textfield" placeholder="公司名称" name="sc.name" size="15" class="form-control"></div>
        <div class="form-group"><select class="form-control form-control-sm ml-1">
            <option value="-1">所有</option>
            <option value="0">企业散客</option>
            <option value="1">企业客户</option>
          </select></div> <label>排序</label> <select class="form-control">
          <option value="0">默认</option>
          <option value="1">协议截止日期（倒序）</option>
        </select> <label>状态</label> <select class="form-control">
          <option value="-1">全部</option>
          <option value="0">停用</option>
          <option value="1">启用</option>
        </select>
        <my-date-picker-stub placeholder="协议截止日期" value="" id="endDate" sizing="sm"></my-date-picker-stub> <label>记录数</label> <select class="form-control">
          <option value="10">10</option>
          <option value="25">25</option>
          <option value="50">50</option>
          <option value="100">100</option>
          <option value="200">200</option>
        </select>
        <div class="form-group"><button type="button" class="btn btn-primary ml-1">查找</button>
          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
          <button type="button" title="清空查找条件" class="btn btn-secondary btn-sm">重置</button></div>
        <router-link-stub to="/customer" class="btn btn-success ml-auto">新建</router-link-stub>
      </form>
    </div>
  </div>
  <table class="table table-striped table-sm small">
    <thead>
      <tr>
        <th>公司名称</th>
        <th>客户编号</th>
        <th>排序拼音</th>
        <th>联系人</th>
        <th>电话</th>
        <th>结算方式</th>
        <th>结算周期</th>
        <th>协议截止日期</th>
        <th>状态</th>
        <th class="text-right"></th>
        <th class="text-right"></th>
      </tr>
    </thead>
    <tbody></tbody>
  </table>
  <nav id="pagination-box" class="float-right">
    <my-pagination-stub rowcount="0" pageno="1" pagetotal="0" name="pagination"></my-pagination-stub>
  </nav>
</div>
`;
